tools/report_build.py

```diff
@@ -88,6 +117,76 @@ class GetEvents(GitHubQLQuery):
         }
     """
 
+    @property
+    def pull_request(self):
+        return self.data["repository"]["pullRequest"]
+
+    @property
+    def events(self):
+        return self.pull_request["timelineItems"]["edges"]
+
+
+class GetCommitComments(GitHubQLQuery):
+    query = """
+        query GetCommitComments($owner: String!,
+                                $name: String!,
+                                $branch: String = null,
+                                $commit: String!,
+                                $count: Int = 10) {
+        repository(owner: $owner, name: $name) {
+          ref(qualifiedName: $branch) {
+            associatedPullRequests(states: OPEN, last: 1) {
+              edges {
+                node {
+                  id
+                  number
+                }
+              }
+            }
+          }
+          object(expression: $commit) {
+            id
+            ... on Commit {
```

Comment:
This, for lack of a better term, [*casts*](https://graphql.github.io/learn/queries/#inline-fragments) the [GitObject](https://developer.github.com/v4/object/repository/#object) to a [Commit](https://developer.github.com/v4/object/commit/) so that you can access its fields. In other words, this is telling the query to use the Commit interface on the Object.

Reply:
Ahh, makes sense